### PR TITLE
[Merged by Bors] - feat(Algebra/Module/Projective): Tensor product of projective modules

### PR DESCRIPTION
**Pull request by erdOne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/Projective.lean [hunk 1]
```diff
@@ -166,6 +167,50 @@ instance (priority := 100) Projective.of_free [Module.Free R P] : Module.Project
...
+    have : Module.Projective R (M ⊗[R₀] (N →₀ R₀)) := by
+      fapply Projective.of_split (R := R) (M := ((M →₀ R) ⊗[R₀] (N →₀ R₀)))
+      · exact (AlgebraTensorModule.map sM (LinearMap.id (R := R₀) (M := N →₀ R₀)))
+      · exact (AlgebraTensorModule.map
+          (Finsupp.total M M R id) (LinearMap.id (R := R₀) (M := N →₀ R₀)))
+      · ext; simp [hsM _]
+    fapply Projective.of_split (R := R) (M := (M ⊗[R₀] (N →₀ R₀)))
+    · exact (AlgebraTensorModule.map (LinearMap.id (R := R) (M := M)) sN)
+    · exact (AlgebraTensorModule.map (LinearMap.id (R := R) (M := M)) (Finsupp.total N N R₀ id))
+    · ext; simp [hsN _]
```
Comment (acmepjz) [inline Mathlib/Algebra/Module/Projective.lean:212]:
```suggestion
    Module.Projective R (M ⊗[R₀] N) := by
  obtain ⟨sM, hsM⟩ := hM
  obtain ⟨sN, hsN⟩ := hN
  have : Module.Projective R (M ⊗[R₀] (N →₀ R₀)) := by
    fapply Projective.of_split (R := R) (M := ((M →₀ R) ⊗[R₀] (N →₀ R₀)))
    · exact (AlgebraTensorModule.map sM (LinearMap.id (R := R₀) (M := N →₀ R₀)))
    · exact (AlgebraTensorModule.map
        (Finsupp.total M M R id) (LinearMap.id (R := R₀) (M := N →₀ R₀)))
    · ext; simp [hsM _]
  fapply Projective.of_split (R := R) (M := (M ⊗[R₀] (N →₀ R₀)))
  · exact (AlgebraTensorModule.map (LinearMap.id (R := R) (M := M)) sN)
  · exact (AlgebraTensorModule.map (LinearMap.id (R := R) (M := M)) (Finsupp.total N N R₀ id))
  · ext; simp [hsN _]
```

File: Mathlib/Algebra/Module/Projective.lean [hunk 2]
```diff
@@ -166,6 +167,50 @@ instance (priority := 100) Projective.of_free [Module.Free R P] : Module.Project
...
+    ∃ (M : Type max u v) (_ : AddCommGroup M) (_ : Module R M) (_ : Module.Free R M)
+      (i : P →ₗ[R] M) (s : M →ₗ[R] P), s.comp i = LinearMap.id :=
+  ⟨fun ⟨i, hi⟩ ↦ ⟨P →₀ R, _, _, inferInstance, i, Finsupp.total P P R id, LinearMap.ext hi⟩,
+    fun ⟨_, _, _, _, i, s, H⟩ ↦ Projective.of_split i s H⟩
+
+/-- A quotient of a projective module is projective iff it is a direct summand. -/
+theorem Projective.iff_split_of_projective [Module.Projective R M] (s : M →ₗ[R] P)
+    (hs : Function.Surjective s) :
+    Module.Projective R P ↔ ∃ i, s ∘ₗ i = LinearMap.id := by
+  refine ⟨fun h ↦ projective_lifting_property _ _ hs, fun ⟨i, H⟩ ↦ Projective.of_split i s H⟩
```
Comment (joelriou) [inline Mathlib/Algebra/Module/Projective.lean:196]: Untested, but `refine` should be unnecessary here:
```suggestion
    Module.Projective R P ↔ ∃ i, s ∘ₗ i = LinearMap.id :=
  ⟨fun h ↦ projective_lifting_property _ _ hs, fun ⟨i, H⟩ ↦ Projective.of_split i s H⟩
```

**Comment by joelriou** [comment]
Thanks!

bors d+

**Comment by erdOne** [comment]
bors merge

**Comment by erdOne** [comment]
Thanks for the review!

**Pull request closed without merging**